### PR TITLE
Rename the confusing neutral annotation in CClosure.

### PR DESCRIPTION
**Pull request by ppedrot**
The `Norm` name was confusing enough to have introduced a few kernel bugs, so it deserved to be changed as suggested in #13274. Contrarily to what it seemingly meant, it was actually standing for neutral terms rather than normal ones. I have kept the 4-letter naming scheme for simplicity and renamed it into `Ntrl`.

### CODE REVIEW

**Comment by ppedrot** [comment]
@SkySkimmer would you assign? It's quite trivial.

File: kernel/cClosure.ml [hunk 1]
```diff
@@ -315,13 +315,13 @@ end = struct
...
+    | 0b0000 -> Ntrl
     | 0b0100 -> Cstr
     | 0b1000 -> Whnf
     | 0b1100 -> Red
     | _ -> assert false
 
-  let[@inline] neutr x = x lor 0b1000 (* Whnf|Norm -> Whnf | Red|Cstr -> Red *)
+  let[@inline] neutr x = x lor 0b1000 (* Whnf|Ntrl -> Whnf | Red|Cstr -> Red *)
 
   let[@inline] set_norm x = x land 0b0011
```
Comment (SkySkimmer) [inline kernel/cClosure.ml:326]: This should be renamed too

File: kernel/cClosure.ml [hunk 2]
```diff
@@ -359,9 +359,9 @@ and finvert = Univ.Instance.t * fconstr array
 
 let fterm_of v = v.term
 let set_norm v = v.mark <- Mark.set_norm v.mark
```
Comment (SkySkimmer) [inline kernel/cClosure.ml:361]: also this

File: kernel/cClosure.ml [hunk 3]
```diff
@@ -315,13 +315,13 @@ end = struct
...
   let[@inline] red_state x = match x land 0b1100 with
-    | 0b0000 -> Norm
+    | 0b0000 -> Ntrl
     | 0b0100 -> Cstr
     | 0b1000 -> Whnf
     | 0b1100 -> Red
     | _ -> assert false
 
-  let[@inline] neutr x = x lor 0b1000 (* Whnf|Norm -> Whnf | Red|Cstr -> Red *)
+  let[@inline] neutr x = x lor 0b1000 (* Whnf|Ntrl -> Whnf | Red|Cstr -> Red *)
```
Comment (SkySkimmer) [inline kernel/cClosure.ml:324]: While we're at it this function's name may be confusing too (called `neutr` but removes `Ntrl`)

**Review by SkySkimmer — CHANGES_REQUESTED** [review]

File: kernel/cClosure.ml (hunk 3 above)
Comment (ppedrot) [reply kernel/cClosure.ml:324]: Suggestion welcome.

**Comment by ppedrot** [comment]
I don't really care about the naming of the `neutr` function, so without further comment I think this PR is ready.

**Review by SkySkimmer — APPROVED** [review]

**Comment by SkySkimmer** [comment]
@coqbot: merge now

**Pull request merged**